### PR TITLE
fix: improved handling of complex yaml in YAMLNodeToJSON

### PR DESCRIPTION
**Pull request by TristanSpeakEasy**
Makes it possible to render yaml that uses anchors and complex map keys

### CODE REVIEW

**Comment by daveshanley** [comment]
Bump those two lines of coverage, and it's good to go.

File: json/json.go [hunk 1]
```diff
@@ -31,28 +30,38 @@ func handleYAMLNode(node *yaml.Node) (any, error) {
...
-	m := orderedmap.New[string, yaml.Node]()
+	v := orderedmap.New[string, any]()
+	for i, n := range node.Content {
+		if i%2 == 0 {
+			continue
+		}
+		keyNode := node.Content[i-1]
+		kv, err := handleYAMLNode(keyNode)
+		if err != nil {
+			return nil, err
```
Comment (daveshanley) [inline json/json.go:48]: Coverage?

File: json/json.go [hunk 2]
```diff
@@ -31,28 +30,38 @@ func handleYAMLNode(node *yaml.Node) (any, error) {
...
+		}
 
-	if err := node.Decode(m); err != nil {
-		return nil, err
-	}
+		if reflect.TypeOf(kv).Kind() != reflect.String {
+			keyData, err := json.Marshal(kv)
+			if err != nil {
+				return nil, err
```
Comment (daveshanley) [inline json/json.go:54]: Coverage?

**Comment by TristanSpeakEasy** [comment]
@daveshanley essential asking for a pass on dealing with those bits of coverage, though if you really do want them covered I guess a way forward is to unit test the internal functions directly (currently I am only testing the public interface with the `package json_test` restriction in the test file) and manually trying to construct an invalid yaml.Node

**Comment by daveshanley** [comment]
I spent about 20 minutes and I could get one line covered, but the other is impossible to test as it's a deep catch. 

Pass granted.

**Pull request merged**